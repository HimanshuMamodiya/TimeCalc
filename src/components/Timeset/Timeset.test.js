import React from 'react'
import ConnectedTimeset, {Timeset} from './Timeset'
import 'jest-enzyme'
import { shallowWithStore, shallowWithState } from 'enzyme-redux'
import { shallow, mount } from 'enzyme'
import { createMockStore } from 'redux-test-utils'
import {Provider} from 'react-redux'

describe('Component - Timeset', () => {
  const date = {start: '12:00', break: '00:30', end: '14:30', duration: '02:00', date: '12/12/2012'}

  it('is wrapped with redux', () => {
    const component = shallowWithState(<ConnectedTimeset time={date} index={1} />, {}).dive()

    expect(component.find(Timeset).length).toBe(1)
  })
})
