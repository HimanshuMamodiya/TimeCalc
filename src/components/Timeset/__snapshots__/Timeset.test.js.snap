// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Timeset renders a date set 1`] = `"<tr class=\\"MuiTableRow-root-4 Timeset-row-2\\"><td class=\\"MuiTableCell-root-9 Timeset-important-3\\">December 12, 2012</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Timeset-hideMobile-1\\">12:00</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Timeset-hideMobile-1\\">14:30</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Timeset-hideMobile-1\\">00:30</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Timeset-important-3\\">02:00</td><td class=\\"MuiTableCell-root-9\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-23 MuiIconButton-root-17\\" type=\\"button\\" aria-label=\\"Edit\\"><span class=\\"MuiIconButton-label-22\\"><svg class=\\"MuiSvgIcon-root-26\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z\\"></path></g></svg></span><span class=\\"MuiTouchRipple-root-32\\"></span></button><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-23 MuiIconButton-root-17\\" type=\\"button\\" aria-label=\\"Delete\\"><span class=\\"MuiIconButton-label-22\\"><svg class=\\"MuiSvgIcon-root-26\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></g></svg></span><span class=\\"MuiTouchRipple-root-32\\"></span></button></td></tr>"`;

exports[`Component - Timeset triggers the delete callback 1`] = `"<tr class=\\"MuiTableRow-root-4 Timeset-row-2\\"><td class=\\"MuiTableCell-root-9 Timeset-important-3\\">December 12, 2012</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Timeset-hideMobile-1\\">12:00</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Timeset-hideMobile-1\\">14:30</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Timeset-hideMobile-1\\">00:30</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Timeset-important-3\\">02:00</td><td class=\\"MuiTableCell-root-9\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-23 MuiIconButton-root-17\\" type=\\"button\\" aria-label=\\"Edit\\"><span class=\\"MuiIconButton-label-22\\"><svg class=\\"MuiSvgIcon-root-26\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z\\"></path></g></svg></span><span class=\\"MuiTouchRipple-root-32\\"></span></button><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-23 MuiIconButton-root-17\\" type=\\"button\\" aria-label=\\"Delete\\"><span class=\\"MuiIconButton-label-22\\"><svg class=\\"MuiSvgIcon-root-26\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></g></svg></span><span class=\\"MuiTouchRipple-root-32\\"></span></button></td></tr>"`;

exports[`Component - Timeset triggers the edit callback 1`] = `"<tr class=\\"MuiTableRow-root-4 Timeset-row-2\\"><td class=\\"MuiTableCell-root-9 Timeset-important-3\\">December 12, 2012</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Timeset-hideMobile-1\\">12:00</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Timeset-hideMobile-1\\">14:30</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Timeset-hideMobile-1\\">00:30</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Timeset-important-3\\">02:00</td><td class=\\"MuiTableCell-root-9\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-23 MuiIconButton-root-17\\" type=\\"button\\" aria-label=\\"Edit\\"><span class=\\"MuiIconButton-label-22\\"><svg class=\\"MuiSvgIcon-root-26\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z\\"></path></g></svg></span><span class=\\"MuiTouchRipple-root-32\\"></span></button><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-23 MuiIconButton-root-17\\" type=\\"button\\" aria-label=\\"Delete\\"><span class=\\"MuiIconButton-label-22\\"><svg class=\\"MuiSvgIcon-root-26\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></g></svg></span><span class=\\"MuiTouchRipple-root-32\\"></span></button></td></tr>"`;
